src/components/Exercise.tsx

```diff
@@ -56,9 +56,10 @@ export const Exercise = ({
   numberOfQuestions, questionNumber, step, exercise, show_all_feedback, scrollToQuestion, ...props
 }: ExerciseWithStepDataProps | ExerciseWithQuestionStatesProps) => {
   const legacyStepRender = 'feedback_html' in step;
+  const questionsRef = React.useRef<Array<HTMLDivElement>>([]);
 
   React.useEffect(() => {
-    const el = document.querySelector(`.openstax-question[data-question-number="${scrollToQuestion}"]`)
+    const el = scrollToQuestion && questionsRef.current[scrollToQuestion];
     if (el) {
```

Comment:
@jivey do you know how I can add code coverage for this line?

Reply:
I tried this in a before block, and with wrapping the expect in `act` and passing scrollToQuestion={1}` it seems to satisfy the coverage:

```
const ref = { current: null };
Object.defineProperty(ref, 'current', {
  get: jest.fn(() => [null, 'element'])
});
jest.spyOn(React, "useRef").mockReturnValue(ref);
```